java/fury-core/src/main/java/org/apache/fury/serializer/CompatibleSerializer.java

```diff
@@ -599,14 +616,15 @@ private Object readFieldValue(FieldResolver.FieldInfo fieldInfo, MemoryBuffer bu
       case ClassResolver.PRIMITIVE_DOUBLE_CLASS_ID:
         return buffer.readFloat64();
       case ClassResolver.STRING_CLASS_ID:
-        return fury.readJavaStringRef(buffer);
+        return fury.readJavaStringRef(buffer, fieldInfo.getFuryFieldAnnotationInfo().nullable);
```

Comment:
Could we add a nullbale field into `fieldInfo` directly to avoid two indirections?